servers/curated-corpus-api/src/admin/resolvers/mutations/Section/index.ts

```diff
@@ -113,3 +114,57 @@ export async function disableEnableSection(
   // snowplow & snowflake to glean & bigquery. we are awaiting a decision
   // on the best path forward for our data pipeline.
 }
+
+/**
+ * Update a Custom Editorial Section.
+ */
+export async function updateCustomSection(
+  parent,
+  { data },
+  context: IAdminContext,
+): Promise<Section> {
+  const { externalId } = data;
+
+  // Load current to perform permission + invariants
+  const existing = await context.db.section.findUnique({
+    where: { externalId },
+  });
+
+  if (!existing) {
+    throw new UserInputError(`Section not found for externalId: ${externalId}`);
+  }
+
+  // Only allow updating custom/manual sections created via this flow
+  if (existing.createSource !== ActivitySource.MANUAL) {
```

Comment:
I don't think we need to check `existingSection.createSource` here. checking for the passed `data.createSource` is sufficient. I would replace this block with: 
```
  // Make sure createSource == MANUAL for now for this mutation
  if (data.createSource !== ActivitySource.MANUAL) {
    throw new UserInputError(
      'Cannot update a custom Section: createSource must remain MANUAL',
    );
  }

```